main.cpp

```diff
@@ -44,6 +46,23 @@ HANDLE WINAPI HookOpenEvent(DWORD dwDesiredAccess, BOOL bInheritHandle, LPCWSTR
 	return TrueOpenEvent(dwDesiredAccess, bInheritHandle, lpName);
 }
 
+BOOL HookFreeLibrary(HMODULE hModule) 
+{
+	WCHAR lpFileName[MAX_PATH];
+	GetModuleFileName(hModule, lpFileName, sizeof(lpFileName));
+	LPCWSTR fileName = PathFindFileNameW(lpFileName);
+
+	WCHAR lpReviveFileName[MAX_PATH];
+	GetModuleFileName((HMODULE)ReviveModule, lpReviveFileName, sizeof(lpReviveFileName));
+	LPCWSTR reviveFileName = PathFindFileNameW(lpReviveFileName);
+
+	if (wcscmp(fileName, reviveFileName) == 0) {
+		return true;
+	}
```

Comment:
You don't need to do all this, this function will be called with the module handle we returned in `HookLoadLibrary`. So you can simply do this here:

```
if (hModule == ReviveModule)
    return true;
```